app/queries/keyset_pagination.rb

```diff
@@ -0,0 +1,130 @@
+module Queries
+  class KeysetPagination
+    attr_reader :client, :order, :key, :count, :previous
+
+    # Initialises the keyset pagination class.
+    # Params:
+    # +client+:: the pagination client to query over
+    # +key+:: +Hash+ a hash containing the pagination key, mapped from presented name to internal name, i.e. { id: "editions.id" }
+    # +order+:: +Symbol+ either :asc for ascending and :desc for descending
+    # +count+:: +Fixnum+ the number of records to return in each page
+    # +after+:: +Array+ the current page to paginate after, an array containing a value for each field in the key
+    def initialize(client, key: nil, order: nil, count: nil, before:, after:)
+      @client = client
+      @key = key || { id: "id" }
+      @order = order || :asc
+      @count = (count || 100).to_i
+
+      if before.present? && after.present?
+        raise "Before and after cannot both be present."
+      end
+
+      if before
+        @previous = before
+        @order = order == :asc ? :desc : :asc
+        @should_reverse = true
+      else
+        @previous = after
+        @should_reverse = false
+      end
```

Comment:
I think you're already thinking about how to present this clearer - so just a nudge to carry on 👍